Chore: Use `--no-cache-dir` flag to `pip` in Dockerfiles, to save space

rajpratik71 opened the pull request:
Using "--no-cache-dir" flag in pip install ,make sure downloaded packages
by pip don't cached on system . This is a best practice which make sure
to fetch from repo instead of using local cached one . Further , in case
of Docker Containers , by restricting caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

sakjur commented on docs/sources/tutorials/stack_guide_graphite.md:
````diff
@@ -169,7 +169,7 @@ Graphite api is a light weight version of graphite-web with only the API compone
 
 ```
-pip install gunicorn graphite-api
+pip install --no-cache-dir gunicorn graphite-api
````
```suggestion
pip install gunicorn graphite-api
```

This is docs meant for end users. I'd rather avoid adding extra arguments and increasing the complexity, especially since the cache is presumably there for a reason when using pip on a regular operating system and not within the context of a Docker container.

aknuds1 replied:
Agree with @sakjur.

sakjur reviewed (changes requested):
Thank you for this PR.

This makes a lot of sense to me except for in documentation when we shouldn't increase the complexity of the instructions for the user. Can we remove that change?

aknuds1 commented on scripts/build/ci-msi-build/msigenerator/Makefile:
```diff
@@ -1,5 +1,5 @@
 
 build:
-	pip3 install -r requirements.txt
+	pip3 install --no-cache-dir -r requirements.txt
```
Please remove this, eventual benefits aren't clear in this case.

Rajpratik71 replied:
done changes as requested

aknuds1 commented:
This branch needs to be merged with master, otherwise CI won't work.

rajpratik71 committed:
optimize size and time using "--no-cache-dir"

Using "--no-cache-dir" flag in pip install ,make sure dowloaded packages
by pip don't cached on system . This is a best practise which make sure
to fetch ftom repo instead of using local cached one . Further , in case
of Docker Containers , by restricing caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

sakjur reviewed (approved):
:+1: LGTM!

aknuds1 commented:
Thanks for contributing to Grafana!
